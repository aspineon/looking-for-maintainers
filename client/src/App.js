import React, { Component } from "react";
import GitHubLogin from "react-github-login";
import { clientId, redirectUri } from "./config/index";
import logo from "./logo.svg";
import "./App.css";

const onSuccess = response => console.log(response);
const onFailure = response => console.error("-ss-", response);

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and safsdfds
          </p>
          <GitHubLogin
            clientId={clientId}
            redirectUri={redirectUri}
            onSuccess={onSuccess}
            onFailure={onFailure}
          />
        </header>
      </div>
    );
  }
}

export default App;
