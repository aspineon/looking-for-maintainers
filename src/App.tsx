import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./all.css";

type AppProps = {
  history: any;
  user: any;
  articles: any;
};

class App extends Component<AppProps> {
  logout = () => {
    localStorage.setItem("user", "");
    this.props.history.push("/");
  };

  render() {
    return (
      <div className="page-content">
        <div className="page-content-container">
          <div className="home">
            <section className="hero-section ">
              <div className="wrapper w960p flex-container flex-container-v">
                <h1
                  data-testid="landing-heading"
                  className="xlbiggest flex-item-center txtcenter"
                >
                  Looking For Maintainers
                </h1>
                <p
                  data-testid="landing-subheading"
                  className="biggest flex-item-center h2-like txtcenter"
                >
                  Web app to find projects to contribute
                </p>
                <p className="biggest flex-item-center h2-like txtcenter"></p>
                <div className="flex-item-center">
                  <div className="mtxl w150p"></div>
                  <Link
                    className="button-primary txtcenter main-cta"
                    to="/explore"
                  >
                    <button className="button-primary txtcenter main-cta">
                      Explore Projects
                    </button>
                  </Link>
                </div>
              </div>
            </section>
            <section className="explanation-section">
              <div className="wrapper w960p flex-container flex-container-v">
                <div className="separator w10"></div>
                <h1 className="xbiggest flex-item-center txtcenter mtxl">
                  Motivation
                </h1>
                <p className="flex-item-center h3-like txtcenter mts">
                  {" "}
                  Most projects on GitHub die because their authors can't find
                  new contributors.
                </p>
                <p className="flex-item-center h3-like txtcenter mts">
                  {" "}
                  &nbsp;{" "}
                </p>
                <p
                  data-testid="landing-text"
                  className="flex-item-center h3-like txtcenter mts"
                >
                  <strong>Looking For Maintainers</strong> will help
                  <strong> project</strong> owners to find
                  <strong> new contributors</strong> and contributors to find
                  exciting projects.
                </p>
                <p className="flex-item-center h3-like txtcenter mts">
                  {" "}
                  &nbsp;{" "}
                </p>
                <div className="flex-item-center">
                  <a
                    className=" button flex-item-center mtxl no-margin"
                    href="https://github.com/manishsaraan/looking-for-maintainers"
                  >
                    {" "}
                    View it on GitHub
                  </a>
                </div>
              </div>
            </section>
            <section className="contact-section mtxl no-margin">
              <div className="wrapper w960p flex-container flex-container-v">
                <div className="separator w10"></div>
                <h1 className="xbiggest flex-item-center txtcenter mtxl">
                  Subscribe Newsletter
                </h1>
                <p className="flex-item-center h3-like txtcenter mts">
                  If you want to get notifications about new projects, please
                  subscribe.
                </p>
                <p className="flex-item-center h3-like txtcenter mts">
                  Zero spam of course, unsubscribe any time.
                </p>
                <div className="contact-container mtxl">
                  <div className="first-page-email">
                    <div className="first-page-email-content">
                      <div className="first-page-email__input-container">
                        <form className="load-email__form" id="contact-form">
                          <div className="load-email__form-item">
                            <input
                              className="load-email-enter"
                              placeholder="Enter your email :)"
                              type="email"
                            />
                          </div>
                          <div className="load-email__form-item txtcenter">
                            <button
                              type="submit"
                              className="load-email-button button-primary"
                            >
                              Subscribe
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </section>
            <footer className="footer-section mtxl">
              <div className="wrapper w960p flex-container flex-container-v txtcenter">
                <div className="h5-like pbs">
                  © Looking For Maintainers • 2020 •
                </div>
              </div>
            </footer>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
