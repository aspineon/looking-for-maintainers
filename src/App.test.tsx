import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

it("renders without crashing", () => {
  const props = {
    history: {},
    user: {},
    articles: {},
  };

  const { container } = render(
    <BrowserRouter>
      <App {...props} />
    </BrowserRouter>
  );
  console.log(container.innerHTML);
});
