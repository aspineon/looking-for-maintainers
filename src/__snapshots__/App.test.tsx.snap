// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Component renders without crashing 1`] = `
<div>
  <div
    class="page-content"
  >
    <div
      class="page-content-container"
    >
      <div
        class="home"
      >
        <section
          class="hero-section "
        >
          <div
            class="wrapper w960p flex-container flex-container-v"
          >
            <h1
              class="xlbiggest flex-item-center txtcenter"
              data-testid="landing-heading"
            >
              Looking For Maintainers
            </h1>
            <p
              class="biggest flex-item-center h2-like txtcenter"
              data-testid="landing-subheading"
            >
              Web app to find projects to contribute
            </p>
            <p
              class="biggest flex-item-center h2-like txtcenter"
            />
            <div
              class="flex-item-center"
            >
              <div
                class="mtxl w150p"
              />
              <a
                class="button-primary txtcenter main-cta"
                href="/explore"
              >
                <button
                  class="button-primary txtcenter main-cta"
                >
                  Explore Projects
                </button>
              </a>
            </div>
          </div>
        </section>
        <section
          class="explanation-section"
        >
          <div
            class="wrapper w960p flex-container flex-container-v"
          >
            <div
              class="separator w10"
            />
            <h1
              class="xbiggest flex-item-center txtcenter mtxl"
            >
              Motivation
            </h1>
            <p
              class="flex-item-center h3-like txtcenter mts"
            >
               
              Most projects on GitHub die because their authors can't find new contributors.
            </p>
            <p
              class="flex-item-center h3-like txtcenter mts"
            >
               
               
               
            </p>
            <p
              class="flex-item-center h3-like txtcenter mts"
            >
              <strong>
                Looking For Maintainers
              </strong>
               will help
              <strong>
                 project
              </strong>
               owners to find
              <strong>
                 new contributors
              </strong>
               and contributors to find exciting projects.
            </p>
            <p
              class="flex-item-center h3-like txtcenter mts"
            >
               
               
               
            </p>
            <div
              class="flex-item-center"
            >
              <a
                class=" button flex-item-center mtxl no-margin"
                href="https://github.com/manishsaraan/looking-for-maintainers"
              >
                 
                View it on GitHub
              </a>
            </div>
          </div>
        </section>
        <section
          class="contact-section mtxl no-margin"
        >
          <div
            class="wrapper w960p flex-container flex-container-v"
          >
            <div
              class="separator w10"
            />
            <h1
              class="xbiggest flex-item-center txtcenter mtxl"
            >
              Subscribe Newsletter
            </h1>
            <p
              class="flex-item-center h3-like txtcenter mts"
            >
              If you want to get notifications about new projects, please subscribe.
            </p>
            <p
              class="flex-item-center h3-like txtcenter mts"
            >
              Zero spam of course, unsubscribe any time.
            </p>
            <div
              class="contact-container mtxl"
            >
              <div
                class="first-page-email"
              >
                <div
                  class="first-page-email-content"
                >
                  <div
                    class="first-page-email__input-container"
                  >
                    <form
                      class="load-email__form"
                      id="contact-form"
                    >
                      <div
                        class="load-email__form-item"
                      >
                        <input
                          class="load-email-enter"
                          placeholder="Enter your email :)"
                          type="email"
                        />
                      </div>
                      <div
                        class="load-email__form-item txtcenter"
                      >
                        <button
                          class="load-email-button button-primary"
                          type="submit"
                        >
                          Subscribe
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <footer
          class="footer-section mtxl"
        >
          <div
            class="wrapper w960p flex-container flex-container-v txtcenter"
          >
            <div
              class="h5-like pbs"
            >
              © Looking For Maintainers • 2020 •
            </div>
          </div>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
